### ISSUE TITLE
Jupyter images are huge

### ISSUE THREAD

**Issue by inc0** [kubeflow/kubeflow#568]
And that's an issue for minikube for example. There are all sorts of things we can do to shrink them.
Couple examples:
* make sure all compile artifacts are gone
* cherry-pick required things with multi-stage builds
* change base image to something lightweight

**Comment by willingc** [comment]
Images are large because there are a ton of dependencies and third party libraries beyond Jupyter notebook being added to the container image.

A minimal Jupyter notebook version is available from jupyter/docker-stacks repo.

 Let's amend this issue title to more accurately reflect the actual issue (lots of installed libraries beyond Jupyter). Thanks!

cc/ @jlewi @inc0

**Comment by willingc** [comment]
Thanks @jlewi

**Comment by inc0** [comment]
@willingc true, issue is on our side rather than jupyter. We create 2 conda envs ~5gig each. I made an attempt to shrink images a bit here #579
